core/src/main/scala/scalaz/Foldable.scala

```diff
@@ -325,6 +325,9 @@ trait Foldable[F[_]]  { self =>
         else seen
     }.reverse
 
+  def distinctBy[A, B: Equal](fa: F[A])(f: A => B): IList[A] =
+    distinctE(fa)(Equal.equalBy(f))
```

Comment:
`distinctBy` could be implemented via `any` :
```scala
  def distinctBy[A, B](fa: F[A])(f: A => B)(implicit B: Equal[B]): IList[A] =
    foldLeft(fa, IList.empty[A]) {
      case (seen, a) =>
        if (!IList.instances.any(seen)(f.andThen(B.equal(f(a), _))))
          a :: seen
        else seen
    }.reverse
```
and then
```scala
def distinctE[A](fa: F[A])(implicit A: Equal[A]): IList[A] =
    distinctBy(fa)(identity)
```
But yeah, it is a very minor variation.